hirundo/dataset_optimization.py

```diff
@@ -287,23 +276,88 @@ def check_run_by_id(run_id: str, retry=0) -> Generator[dict, None, None]:
                 OptimizationDataset._read_csv_to_df(data)
                 yield data
         if not last_event or last_event["data"]["state"] == "PENDING":
-            OptimizationDataset.check_run_by_id(run_id, retry + 1)
+            OptimizationDataset._check_run_by_id(run_id, retry + 1)
 
-    def check_run(self) -> Generator[dict, None, None]:
+    def check_run_by_id(
+        self, run_id: str, stop_on_manual_approval: bool = False
+    ) -> pd.DataFrame:
         """
-        Check the status of the current active instance's run.
+        Check the status of a run given its ID
 
-        This generator will produce values to show progress of the run.
+        Args:
+            run_id: The `run_id` produced by a `run_optimization` call
+            stop_on_manual_approval: If True, the function will return an empty DataFrame if the run is awaiting manual approval
```

Comment:
Didn't we say that should be an environment variable or something? Why is it part of the Python API?
Meaning, do you envision a client explicitly using `stop_on_manual_approval=True` to indicate that he awaits for a manual approval? It's a little different than what we envisioned initially regarding the "defensive" nature of the "Shmuli" button. I'd like to hear your thoughts :)

Reply:
Got it. Makes sense :)